examples/EF_QAE/main.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import numpy as np
+from qibo.models import Circuit
+from qibo import hamiltonians, gates, models, matrices
+from qibo.hamiltonians import Hamiltonian
+from scipy.optimize import minimize
+from sklearn.datasets import load_digits
+import sys
+import argparse
+
+
+def main(layers, autoencoder, example):
+    
+    def encoder_hamiltonian_simple(nqubits, ncompress):
+        """Creates the encoding Hamiltonian.
+        Args:
+            nqubits (int): total number of qubits.
+            ncompress (int): number of discarded/trash qubits.
+
+        Returns:
+            Encoding Hamiltonian.
+        """
+        m0 = hamiltonians.Z(ncompress, numpy=True).matrix
+        m1 = np.eye(2 ** (nqubits - ncompress), dtype=m0.dtype)
+        ham = hamiltonians.Hamiltonian(nqubits, np.kron(m1, m0))
+        return 0.5 * (ham + ncompress)
+    
+    def ansatz_QAE(theta):
+        """Creates the variational quantum circuit for QAE.
+        Args:
+            theta (array or list): values of the parameters.
+
+        Returns:
+            Quantum circuit.
+        """
+        circuit = models.Circuit(nqubits)
+        index = 0
+        for l in range(layers):
+            for q in range(nqubits):
+                circuit.add(gates.RY(q, theta[index]))
+                index+=1
+            circuit.add(gates.CZ(5, 4)); circuit.add(gates.CZ(5, 3))
+            circuit.add(gates.CZ(5, 1)); circuit.add(gates.CZ(4, 2)); circuit.add(gates.CZ(4, 0))
```

Comment:
Please do not use multi statements per line.

Reply:
I know that this is fixed now, but just for future reference note that it is possible to do it in a single line/command as:
```Python
circuit.add([gates.CZ(5, 1), gates.CZ(4, 2), gates.CZ(4, 0)])
```

Reply:
Fine, good to know. Thanks.